dotnet/src/Connectors/Connectors.GoogleVertexAI/Extensions/GoogleAIServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,140 @@
+    public static IKernelBuilder AddGoogleAIGeminiTextGeneration(
+        this IKernelBuilder builder,
+        string modelId,
+        string apiKey,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(modelId);
+        Verify.NotNull(apiKey);
+
+        builder.Services.AddKeyedSingleton<ITextGenerationService>(serviceId, (serviceProvider, _) =>
+            new GoogleAIGeminiTextGenerationService(
+                model: modelId,
+                apiKey: apiKey,
+                httpClient: HttpClientProvider.GetHttpClient(httpClient, serviceProvider),
+                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+        return builder;
+    }
+
+    /// <summary>
+    /// Add Google AI Gemini Text Generation service to the specified service collection.
+    /// </summary>
+    /// <param name="services">The service collection to add the Gemini Text Generation service to.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="apiKey">The API key for authentication Gemini API.</param>
+    /// <param name="serviceId">Optional service ID.</param>
+    /// <returns>The updated service collection.</returns>
+    public static IServiceCollection AddGoogleAIGeminiTextGeneration(
+        this IServiceCollection services,
+        string modelId,
+        string apiKey,
+        string? serviceId = null)
+    {
+        Verify.NotNull(services);
+        Verify.NotNull(modelId);
+        Verify.NotNull(apiKey);
+
+        return services.AddKeyedSingleton<ITextGenerationService>(serviceId, (serviceProvider, _) =>
+            new GoogleAIGeminiTextGenerationService(
+                model: modelId,
+                apiKey: apiKey,
+                httpClient: HttpClientProvider.GetHttpClient(serviceProvider),
+                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+    }
+
+    /// <summary>
+    /// Add Google AI Gemini Chat Completion and Text Generation services to the kernel builder.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="apiKey">The API key for authentication Gemini API.</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <param name="httpClient">The optional custom HttpClient.</param>
+    /// <returns>The updated kernel builder.</returns>
+    public static IKernelBuilder AddGoogleAIGeminiChatCompletion(
+        this IKernelBuilder builder,
+        string modelId,
+        string apiKey,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(modelId);
+        Verify.NotNull(apiKey);
+
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
```

Comment:
Not a very fan of adding as singleton for all IServiceCollection extensions by default, but I'm aware this is how we do in other connectors. 👍🏻 

Looks good

Reply:
I think we can leave it as is also in other connectors. Because ChatCompletionService implementations are public classes, so consumer always can create own extension do DI.